lib/Sema/CSSimplify.cpp

```diff
@@ -272,6 +272,27 @@ static bool isCodeCompletionTypeVar(Type type) {
   return false;
 }
 
+/// Returns \c true if \p type contains a type variable that has been created
+/// for the code completion token.
+static bool containsCodeCompletionTypeVar(Type type) {
+  class CodeCompletionTypeFinder : public TypeWalker {
+  public:
+    bool ContainsCodeCompletionTypeVar = false;
+
+    Action walkToTypePre(Type T) override {
+      if (isCodeCompletionTypeVar(T)) {
+        ContainsCodeCompletionTypeVar = true;
+        return Action::Stop;
+      } else {
+        return Action::Continue;
+      }
+    }
+  };
+  CodeCompletionTypeFinder CompletionTypeFinder;
+  type.walk(CompletionTypeFinder);
+  return CompletionTypeFinder.ContainsCodeCompletionTypeVar;
```

Comment:
The easier way of doing this would be to use `type.findIf([&](Type type) { return isCodeCompetionTypeVar(type); });`

Reply:
Yes, that’s definitely cleaner